File: ScaleformUI_Csharp/Panels/UIMenuPanel.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScaleformUI
{
    public class UIMenuPanel
    {
        public UIMenuPanel()
        {
        }
        public virtual bool Selected { get; set; }
        public virtual bool Enabled { get; set; }
        public virtual void UpdateParent()
        {
        }
        public void SetParentItem(UIMenuItem item)
        {
            ParentItem = item;
        }

        public UIMenuItem ParentItem { get; set; }

    }
}

File: ScaleformUI_V8/src/menus/UIMenu/panels/uimenupanel.ts
import { UIMenuItem } from "../items/uimenuitem";

export abstract class UIMenuPanel {
    public Selected: boolean = false
    public Enabled: boolean = true
    public UpdateParent(){

    }

    public SetParentItem(item: UIMenuItem) {
        this.ParentItem = item;
    }

    public ParentItem: UIMenuItem;
}